src/NServiceBus.CustomChecks/CustomChecksRegistry.cs

```diff
@@ -14,30 +15,26 @@ public void AddScannedTypes(IEnumerable<Type> availableTypes)
 
             foreach (var checkType in availableTypes.Where(IsCustomCheck))
             {
-                var wrapperType = typeof(CustomCheckWrapper<>).MakeGenericType(checkType);
-                var wrapper = (ICustomCheckWrapper)Activator.CreateInstance(wrapperType)!;
-                wrappers.Add(wrapper);
+                _ = AddCustomCheckMethodInfo.MakeGenericMethod(checkType).Invoke(this, []);
```

Comment:
This thought is outside the scope of this PR, but this might be an interesting "constrained test" repo where we could enable trimming warnings and then figure out how to annotate cases like this where the AddScannedTypes method would never be called in a scenario where trimming was enabled.